### PR TITLE
feat(cdn): remove storage unit

### PR DESCRIPTION
**Pull request by sguiheux**
1. Description
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

File: engine/sql/cdn/013_storage_unit_to_delete.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- +migrate Up
+ALTER TABLE storage_unit ADD COLUMN IF NOT EXISTS to_delete BOOLEAN DEFAULT false;
+
+-- +migrate Down
+ALTER TABLE  storage_unit DROP COLUMN to_delete;
```
Comment (richardlt) [inline engine/sql/cdn/013_storage_unit_to_delete.sql:5]:
```suggestion
ALTER TABLE storage_unit DROP COLUMN to_delete;
```

File: engine/cdn/storage/dao.go [hunk 2]
```diff
@@ -248,6 +248,15 @@ func LoadAllItemUnitsIDsByItemID(db gorp.SqlExecutor, itemID string) ([]string,
 	return IDs, nil
 }
 
+func LoadAllItemUnitsIDsByUnitID(db gorp.SqlExecutor, unitID string, offset, limit int64) ([]string, error) {
+	var IDs []string
+	query := "SELECT id FROM storage_unit_item WHERE unit_id = $1 to_delete = false ORDER BY id ASC OFFSET $2 LIMIT $3"
```
Comment (richardlt) [inline engine/cdn/storage/dao.go:253]:
```suggestion
	query := "SELECT id FROM storage_unit_item WHERE unit_id = $1 AND to_delete = false ORDER BY id ASC OFFSET $2 LIMIT $3"
```

File: cli/cdsctl/admin_cdn_unit.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/ovh/cds/cli"
+	"github.com/ovh/cds/sdk"
+)
+
+var adminCdnUnitCmd = cli.Command{
+	Name:  "unit",
+	Short: "Manage CDS CDN unit",
+}
+
+func adminCdnUnit() *cobra.Command {
```
Comment (richardlt) [inline cli/cdsctl/admin_cdn_unit.go:16]: Could be nice to have a command to list existing unit and get their ids.

File: engine/cdn/cdn_gc.go [hunk 4]
```diff
@@ -85,6 +98,63 @@ func (s *Service) markUnitItemToDeleteByItemID(ctx context.Context, itemID strin
 	return n, sdk.WithStack(tx.Commit())
 }
 
+func (s *Service) cleanStorageUnit(ctx context.Context) error {
+	units, err := storage.LoadAllUnits(ctx, s.Mapper, s.mustDBWithCtx(ctx))
```
Comment (richardlt) [inline engine/cdn/cdn_gc.go:102]: Direct load unit to delete ?

**Review by richardlt — CHANGES_REQUESTED** [review]
I think than rather then checking for unit to delete with a ticker and because dropping a unit is not a usual action. It could be better to have a some commands:
- one to "disable" the unit (off sync, stop loading data from this unit).
- a another to mark all unit_items to delete. 
- a last one to direct delete the item that fails if there are still unit_items.

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15665.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15669.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15674.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#15674.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✘

**Review by fsamin — APPROVED** [review]

**Review by richardlt — APPROVED** [review]

**Pull request merged**